website/static/js/comment.js

```diff
@@ -200,10 +210,18 @@ var BaseComment = function() {
 
 };
 
-BaseComment.prototype.handleEditableUpdate = function(element, underMaxLength, charLimit) {
+BaseComment.prototype.handleEditableUpdate = function(element) {
     var self = this;
-    self.underMaxLength(underMaxLength);
-    self.errorMessage(underMaxLength ? '' : 'Exceeds character limit. Please reduce to ' + charLimit + ' characters or less.');
+    var $element = $(element);
+    var charLimit = $element.attr('maxlength');
+    var inputTextLength = $element[0].innerText.length || 0;
+    var showLength = inputTextLength === 1 ? 1 : inputTextLength - 1;
+    // + 1 to account for the <br> that is added to the end of the contenteditable content
+    // <br> is necessary for the return key to function properly
+    var underOrEqualMaxLength = inputTextLength <= parseInt(charLimit) + 1 || charLimit == undefined;  // jshint ignore: line
+    self.currentCount(showLength);
+    self.underMaxLength(underOrEqualMaxLength);
+    self.errorMessage(underOrEqualMaxLength ? '' : 'Exceeds character limit. Please reduce to ' + charLimit + ' characters or less.');
```

Comment:
This error message should be unnecessary. Disable the save button when you go over the limit. The character count should be a standard-enough control that it shouldn't require explanation, and it's right by the save button. You could have this as a tooltip for the disabled save button, though.